src/goto-symex/field_sensitivity.cpp

```diff
@@ -0,0 +1,346 @@
+/*******************************************************************\
+
+Module: Field-sensitive SSA
+
+Author: Michael Tautschnig, mt@eecs.qmul.ac.uk
+
+\*******************************************************************/
+
+#include <util/simplify_expr.h>
+#include <util/std_expr.h>
+#include <util/arith_tools.h>
+
+#include <ansi-c/c_types.h>
+
+#include "symex_target.h"
+#include "goto_symex_state.h"
+
+#include "field_sensitivity.h"
+
+/*******************************************************************\
+
+Function: field_sensitivityt::operator()
+
+  Inputs:
+
+ Outputs:
+
+ Purpose:
+
+\*******************************************************************/
+
+void field_sensitivityt::operator()(
```

Comment:
Since this is being called by name as "operator()" quite a lot, suggest giving it a real name